### ISSUE TITLE
introduce flake8 lint checking to help avoid programming errors and consistent styling

### ISSUE THREAD

**Issue by marc-portier** [brianhealey/pyamplipi#15]
Applying flake8 on the current codebase gives a nice report:

```
$ flake8 . --count --exit-zero --max-complexity=10 --max-line-length=132 --statistics --exclude venv,.venv,.eggs,.tox,.git,__pycache__,*.pyc
./pyamplipi/amplipi.py:1:1: F401 'json' imported but unused
./pyamplipi/amplipi.py:110:44: F541 f-string is missing placeholders
./pyamplipi/cli.py:9:1: F401 'pyamplipi.models.ZoneUpdate' imported but unused
./pyamplipi/cli.py:9:1: F401 'pyamplipi.models.ZoneUpdateWithId' imported but unused
./pyamplipi/models.py:35:1: E302 expected 2 blank lines, found 1
./setup.py:22:9: F841 local variable 'reqs' is assigned to but never used
1     E302 expected 2 blank lines, found 1
3     F401 'json' imported but unused
1     F541 f-string is missing placeholders
1     F841 local variable 'reqs' is assigned to but never used
6
```

Much of this is strictly "style conformity" - but the last one actually helps discovering a recently introduced bug.

If you guys think it useful I would like to prepare a PR that 
- adds flake8 to `requirements-dev.txt` + update `setup.py` to be aware of it (as already hinted at in #6) 
- fixes the current remarked issues in the code 
- introduce a github-ci-workflow to run the flake8 (preferably via `make lint` goverened by introducing a `Makefile`)

**Comment by linknum23** [comment]
This would be a great start to a ci pipeline.

**Comment by marc-portier** [comment]
cool - for convenience I will just tackle this together with #16 and provide one single PR - that will also include the fix-after-merge for #6 and #10

**Comment by linknum23** [comment]
Awesome. I was just about to start that. Thanks for all the help!
